### PR TITLE
fix vbft shard bug

### PR DESCRIPTION
**Pull request by qiluge**
1. fix shard state serialize and deserialize bug;
2. fix shard stake bug while peer init stake;
3. fix fetch parent height at shard block pool bug;
4. add version while construct block header;

### CODE REVIEW

File: smartcontract/service/native/shardmgmt/shardmgmt.go [hunk 1]
```diff
@@ -116,6 +117,8 @@ func ShardMgmtInit(native *native.NativeService) ([]byte, error) {
 			ShardID:             native.ShardID,
 			GenesisParentHeight: native.Height,
 			State:               shardstates.SHARD_STATE_ACTIVE,
+			Config:              &shardstates.ShardConfig{VbftConfigData: &config.VBFTConfig{}},
```
Comment (Honglei-Cong) [inline smartcontract/service/native/shardmgmt/shardmgmt.go:120]: an empty shard config?
why not nil?

File: smartcontract/service/native/shardmgmt/shardmgmt.go [hunk 2]
```diff
@@ -380,37 +384,29 @@ func JoinShard(native *native.NativeService) ([]byte, error) {
 		if err != nil {
 			return utils.BYTE_FALSE, fmt.Errorf("JoinShard: failed, err: %s", err)
 		}
-		if rootChainPeerItem.TotalPos < params.StakeAmount {
+		if rootChainPeerItem.InitPos < params.StakeAmount && rootChainPeerItem.TotalPos < params.StakeAmount {
```
Comment (Honglei-Cong) [inline smartcontract/service/native/shardmgmt/shardmgmt.go]: only need to check the InitPos?

File: smartcontract/service/native/shardmgmt/shardmgmt.go [hunk 3]
```diff
@@ -380,37 +384,29 @@ func JoinShard(native *native.NativeService) ([]byte, error) {
...
-			peerStakeInfo.NodeType = shardstates.CONSENSUS_NODE
-		} else {
-			return utils.BYTE_FALSE, fmt.Errorf("JoinShard: unsupport consensus type")
-		}
-		peerStakeInfo.Index = uint32(len(shard.Peers) + 1)
 		shard.Peers[strings.ToLower(params.PeerPubKey)] = peerStakeInfo
+		if shard.Config.VbftConfigData.Peers == nil {
+			shard.Config.VbftConfigData.Peers = make([]*config.VBFTPeerStakeInfo, 0)
+		}
+		shard.Config.VbftConfigData.Peers = append(shard.Config.VbftConfigData.Peers, &config.VBFTPeerStakeInfo{
```
Comment (Honglei-Cong) [inline smartcontract/service/native/shardmgmt/shardmgmt.go]: suggest ：
```
peerStakeInfo := &config.VBFTPeerStakeInfo{...}
shard.Config.VbftConfigData.Peers = append(shard.Config.VbftConfigData.Peers, peerStakeInfo)
```

File: wallet-consensus.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (Honglei-Cong) [inline wallet-consensus.json]: why add this file?

**Review by Honglei-Cong — CHANGES_REQUESTED** [review]

File: smartcontract/service/native/shardmgmt/shardmgmt.go (hunk 1 above)
Comment (qiluge) [reply smartcontract/service/native/shardmgmt/shardmgmt.go:120]: in case of nil pointer error while serialize

**Review by Honglei-Cong — APPROVED** [review]

**Pull request merged**